### PR TITLE
Enable more Ruff lints & fix issues

### PR DESCRIPTION
**Pull request by akx**
# What does this PR do?

Follows up on #2400. 
Sibling of https://github.com/huggingface/trl/pull/1300 and https://github.com/huggingface/peft/pull/1421

I didn't enable the full `B` family yet, because there's still about 90 errors left if I do, and best keep these PRs somewhat reviewable...

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

cc @muellerzr

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
+ignore = [
+    "E501", # Line length (handled by ruff-format)
+    "E741", # Ambiguous variable name
+    "W605", # Invalid escape sequence
+    "UP007", # X | Y type annotations
+]
+
+[tool.ruff.lint.per-file-ignores]
+"__init__.py" = [
+    "F401",  # Ignore seemingly unused imports (they're meant for re-export)
```
Comment (BenjaminBossan) [inline pyproject.toml:23]: This could be handled by defining `__all__` in the corresponding modules, but I'm fine with simply ignoring those imports.

**Review by BenjaminBossan — CHANGES_REQUESTED** [review]
Thanks for extending the use of ruff in this repo. The changes look good, I just found one instance which I think is better fixed manually. Please take a look.

File: pyproject.toml (hunk 1 above)
Comment (akx) [reply pyproject.toml:23]: I had a commit for that in here too, but it added +300 lines, so I decided to skip it. Can easily add in a subsequent PR.

File: src/accelerate/utils/megatron_lm.py [hunk 2]
```diff
@@ -845,8 +844,8 @@ def initialize(accelerator, extra_args_provider=None, args_defaults={}):
         if getattr(args, key, None) is not None:
             if args.rank == 0:
                 print(
-                    "WARNING: overriding default arguments for {key}:{v} \
-                        with {key}:{v2}".format(key=key, v=getattr(args, key), v2=value),
+                    f"WARNING: overriding default arguments for {key}:{getattr(args, key)} \
```
[the remark this answers is not in the document]
Comment (akx) [reply src/accelerate/utils/megatron_lm.py:847]: Yeah, the autofix didn't do anything about the `\` continuation. Will reformat!

**Comment by akx** [comment]
Gentle ping @BenjaminBossan :)

**Review by BenjaminBossan — APPROVED** [review]
Thanks for the adjustments, LGTM. Leaving it up to @muellerzr to merge.

And please always ping when the PR is ready for the next review round.

File: src/accelerate/utils/megatron_lm.py [hunk 3]
```diff
@@ -845,8 +844,8 @@ def initialize(accelerator, extra_args_provider=None, args_defaults={}):
         if getattr(args, key, None) is not None:
             if args.rank == 0:
                 print(
-                    "WARNING: overriding default arguments for {key}:{v} \
-                        with {key}:{v2}".format(key=key, v=getattr(args, key), v2=value),
+                    f"WARNING: overriding default arguments for "
```
Comment (BenjaminBossan) [inline src/accelerate/utils/megatron_lm.py:847]: the `f` on this line is not needed

**Review by muellerzr — APPROVED** [review]
Thanks for bringing us to 2.x.x for ruff! Can you run `make style; make quality` once more? This should fix the last failure case (in `megatron`)

**Comment by akx** [comment]
> Can you run `make style; make quality` once more? 

Rebased (and fixed the couple new `UP` complaints that popped up from `pippy` examples), done.

**Comment by BenjaminBossan** [comment]
I did a re-check after the latest changes, still LGTM.

**Pull request merged**